src/parser.ts

```diff
@@ -3478,6 +3478,33 @@ export class Parser extends DiagnosticEmitter {
     return null;
   }
 
+  private getRecursiveDepthForTypeDeclaration(
+    identifierName: string,
+    type: TypeNode,
+    depth: i32 = 0
+  ): i32 {
+    switch (type.kind) {
+      case NodeKind.NAMEDTYPE: {
+        if ((<NamedTypeNode>type).name.identifier.text == identifierName) {
+          return depth;
+        }
```

Comment:
Should this perhaps also check type parameters if `type` is generic? For example
```ts
type A = B<A>;
```

Reply:
DO you mean `type A = Array<A>`? It should be an `Not implementer` error while it support in TS

Reply:
For example, yes. Seems this is not checked yet?

Reply:
Yeah, This was not checked. Fix this scenario as well